Coloring Book: fix font conflict and move JS files (#116)
MicrosoftEdge/Demos: pwa-install-element/README.md

> # Install Element - &lt;Install&gt;

Reviewer: ```suggestion
# Install element demos - `<install>`
```

This directory contains demos that showcase the use of the [install element](https://aka.ms/InstallElement), a new HTML element under development by the Microsoft Edge team to allow web contents to declaratively install other web apps. The element is currently supported on Windows, MacOS, Linux, and ChromeOS.

## Demos

> ➡️ **[Open the demo](https://liahiscock.github.io/PWA/WebInstall/element-store.html)** ⬅️

Reviewer: Can we move demos in this directory instead of pointing to an external link? We're trying to consolidate all our demos in this repo, so anyone can update them in one place.

It's ok to do this in a later PR.

> ## How to Use It

Reviewer: ```suggestion
## How to use the `<install>` element
```

Please use sentence casing for all headings.
Try and avoid using `it`. In this instance, it was confusing: how to use what? The demo, or the API?

> 

Reviewer: Might be good to start with a support section:

```suggestion

### Detect support

if ('HTMLInstallElement' in window) {
  // The <install> element is supported.
} else {
  // The <install> element is not supported.
}

```

### To install the currently loaded document

#### &lt;install&gt;&lt;/install&gt; Requirements
* The current loaded document must link to a manifest file.

> * That manifest file must have an `id` field defined.

Reviewer: ```suggestion
### To install the current document

To install the currently loaded document:

* The document must link to a manifest file.
* The manifest file must have an `id` field defined.
```

```javascript
/* Current Document: no attributes required */
<install id="install-button"></install>
```
### To install a background document (anything that's not the current document)

**&lt;install installurl=...&gt;&lt;/install&gt; Requirements:**
* The document at `installurl` must link to a manifest file.

> * The manifest file must have an `id` field defined.

Reviewer: ```suggestion
### To install another document

To install a document that's not the current document, also known as a _background_ document, use either the `installurl` attribute, or both the `installurl` and `manifestid` attributes together:

To use the `installurl` attribute:

* The document at `installurl` must link to a manifest file.
* The manifest file must have an `id` field defined.
```

```javascript
/*Background Document: 1-attribute*/
<install installurl="https://foo.com"
         id="install-button"></install>
```
**&lt;install installurl=... manifestid=...&gt;&lt;/install&gt; Requirements:**
* The document at  `installurl` must link to a manifest file.
* `manifestid` must match the computed id after parsing the manifest.

> > The computed ID can be found by navigating to `installurl`, opening F12 developer tools > Application tab > Installability section > Computed App ID

Reviewer: ```suggestion

To use the `installurl` and `manifestid` attributes together:

* The document at  `installurl` must link to a manifest file.
* The value of the `manifestid` attribute must match the computed id after parsing the manifest.

You can find the computed ID by going to **Application** > **Manifest** > **Identity** > **Computed App ID** in Microsoft Edge DevTools.
```

```javascript
/*Background Document: 2-attributes*/
<install installurl="https://foo.com"
         manifestid="https://foo.com/someid"
         id="install-button">
</install>
```

> ### Error handling

Reviewer: ```suggestion
### Handle installation success and errors

To handle the result of the web app installation process, use the `promptaction`, `promptdismiss`, and `validationstatuschanged` events:
```

```javascript
if ('HTMLInstallElement' in window) {
    let button = document.getElementById('install-button');

    // Same as kSuccess in navigator.install.
    button.addEventListener('promptaction', (event) => {
        console.log(`Install succeeded`);
    });

    // Same as kAbortError in navigator.install.
    button.addEventListener('promptdismiss', (event) => {
        console.log(`Install failed`);
    });

    // Same as kDataError in navigator.install.
    button.addEventListener('validationstatuschanged', (event) => {
        if (event.target.invalidReason ===
                'install_data_invalid') {
            console.log(event.target.invalidReason);
        }

>     });

Reviewer: ```suggestion
    const button = document.getElementById('install-button');

    // Listen to the promptaction event to know if the installation succeeded.
    button.addEventListener('promptaction', (event) => {
        console.log(`Install succeeded`);
    });

    // Listen to the promptdismiss event to know if the installation failed.
    button.addEventListener('promptdismiss', (event) => {
        console.log(`Install failed`);
    });

    // Listen to the validationstatuschanged event to detect invalid installation data.
    button.addEventListener('validationstatuschanged', (event) => {
        if (event.target.invalidReason === 'install_data_invalid') {
            console.log(event.target.invalidReason);
        }
    });
```

} else {
  console.warn('HTMLInstallElement not supported');
}
```

> ## Try it with Origin Trials!

Reviewer: ```suggestion
## Test the feature in product with origin trials
```

> The install element is currently available as an [Origin Trial](https://developer.chrome.com/docs/web-platform/origin-trials/) in Chrome and Microsoft Edge versions 148-153. This allows you to use the feature on your production site and provide valuable feedback to browser vendors before it's finalized.

Reviewer: ```suggestion
The install element is currently available as an [Origin Trial](https://learn.microsoft.com/microsoft-edge/origin-trials/) in Microsoft Edge and Chrome versions 148 to 153. This allows you to use the feature on your production site and provide valuable feedback to browser vendors before it's finalized.
```

> To participate, you'll need to:

Reviewer: ```suggestion
To use the install element on your production site, participate in the origin trial:
```

> >TODO(ADD REAL OT PORTAL LINK)

Reviewer: TODO

1. **Register for the Origin Trial:** [Install element registration page link](https://developer.chrome.com/origintrials/#/view_trial/XXXXXXXXXX)

> 2. **Add the Origin Trial Token:** Once you have your token, add it to your pages via a `<meta>` tag or an HTTP header.

Reviewer: ```suggestion
1. **Register for the origin trial:** [Install element registration page link](https://developer.chrome.com/origintrials/#/view_trial/XXXXXXXXXX)
2. **Add the origin trial token:** Once you have the origin trial registration token, add the token to your pages by using a `<meta>` tag or an HTTP header. To learn more, see [Using the origin trial token at your website](https://learn.microsoft.com/microsoft-edge/origin-trials/#using-the-origin-trial-token-at-your-website).
```

```html
<!-- Example via a meta tag -->
<meta http-equiv="origin-trial" content="YOUR_TOKEN_HERE">
```

> See [Origin Trials Guide for Web Developers](https://github.com/GoogleChrome/OriginTrials/blob/gh-pages/developer-guide.md) to learn more about Origin Trials.

Reviewer: ```suggestion
```

> ## Provide Feedback

Reviewer: ```suggestion
## Provide feedback
```

> Your feedback is crucial to the development of this feature. If you encounter any issues, have suggestions, or want to share how you're using the Web Install API, please:

Reviewer: ```suggestion
Your feedback is crucial to the development of this feature. Please share feedback to:

* Report any issue you encountered.
* Share improvement suggestions.
* Share how you're using the `<install>` element.
```

> **Log an issue here:** [Install element Feedback Link](https://github.com/WICG/install-element/issues/new?template=install-element-ot-feedback.md)

Reviewer: ```suggestion
To share feedback, [open a new issue on the WICG/install-element repo](https://github.com/WICG/install-element/issues/new?template=install-element-ot-feedback.md)
```

# Install element demos - `<install>`

This directory contains demos that showcase the use of the [install element](https://aka.ms/InstallElement), a new HTML element under development by the Microsoft Edge team to allow web contents to declaratively install other web apps. The element is currently supported on Windows, MacOS, Linux, and ChromeOS.

## Demos

> * ➡️ **[The <install> Element Store](https://liahiscock.github.io/PWA/WebInstall/element-store.html)** ⬅️

Reviewer: ```suggestion
* ➡️ **[The `<install>` Element Store](https://liahiscock.github.io/PWA/WebInstall/element-store.html)** ⬅️
```

```javascript
<install installurl="https://foo.com" manifestid="https://foo.com/someid" id="install-button"></install>
```

### Handle installation success and errors

To handle the result of the web app installation process, use the `promptaction`, `promptdismiss`, and `validationstatuschanged` events:

```javascript
if ('HTMLInstallElement' in window) {
  const button = document.getElementById('install-button');

  // Listen to the promptaction event to know if the installation succeeded.
  button.addEventListener('promptaction', (event) => {
    console.log(`Install succeeded`);
  });

  // Listen to the promptdismiss event to know if the installation failed.
  button.addEventListener('promptdismiss', (event) => {
    console.log(`Install failed`);
  });

  // Listen to the validationstatuschanged event to detect invalid installation data.
  button.addEventListener('validationstatuschanged', (event) => {
    if (event.target.invalidReason === 'install_data_invalid') {
      console.log(event.target.invalidReason);
    }
  });
} else {
  console.warn('HTMLInstallElement not supported');
}
```

> ## Test the feature in production with origin trials

Reviewer: Can we also add another section before this one like "Test the feature locally". Many devs will not want to register to the OT but might still want to test locally by enabling the flag.

1. **Register for the origin trial:** [Install element registration page link](https://developer.chrome.com/origintrials/#/view_trial/XXXXXXXXXX)
2. **Add the origin trial token:** Once you have the origin trial registration token, add the token to your pages by using a `<meta>` tag or an HTTP header. To learn more, see [Using the origin trial token at your website](https://learn.microsoft.com/microsoft-edge/origin-trials/#using-the-origin-trial-token-at-your-website).

```html
<!-- Example via a meta tag -->
<meta http-equiv="origin-trial" content="YOUR_TOKEN_HERE">
```

## Provide feedback

Your feedback is crucial to the development of this feature. Please share feedback to:

* Report any issue you encountered.
* Share improvement suggestions.
* Share how you're using the `<install>` element.

To share feedback, [open a new issue on the WICG/install-element repo](https://github.com/WICG/install-element/issues/new?template=install-element-ot-feedback.md)

We look forward to hearing from you!

## Further Reading and References
* [Explainer](https://aka.ms/InstallElement)
* [Chrome Platform Status Entry](https://chromestatus.com/feature/5152834368700416)

> * [Origin Trials Guide for Web Developers](https://github.com/GoogleChrome/OriginTrials/blob/gh-pages/developer-guide.md)

Reviewer: ```suggestion
## See also

* [Explainer](https://aka.ms/InstallElement)
* [Chrome Platform Status Entry](https://chromestatus.com/feature/5152834368700416)
* [Test experimental APIs and features by using origin trials](https://learn.microsoft.com/microsoft-edge/origin-trials/#using-the-origin-trial-token-at-your-website)
```

```javascript
<install installurl="https://foo.com" manifestid="https://foo.com/someid" id="install-button"></install>
```

### Handle installation success and errors

To handle the result of the web app installation process, use the `promptaction`, `promptdismiss`, and `validationstatuschanged` events:

```javascript
if ('HTMLInstallElement' in window) {
  const button = document.getElementById('install-button');

  // Listen to the promptaction event to know if the installation succeeded.
  button.addEventListener('promptaction', (event) => {
    console.log(`Install succeeded`);
  });

  // Listen to the promptdismiss event to know if the installation failed.
  button.addEventListener('promptdismiss', (event) => {
    console.log(`Install failed`);
  });

  // Listen to the validationstatuschanged event to detect invalid installation data.
  button.addEventListener('validationstatuschanged', (event) => {
    if (event.target.invalidReason === 'install_data_invalid') {
      console.log(event.target.invalidReason);
    }
  });
} else {
  console.warn('HTMLInstallElement not supported');
}
```

> ## Test the feature locally
> 
> The install element can be tested via a command line flag, which is available in Chromium-based browsers (such as Microsoft Edge) starting with **version 148**.
> 
> 1. In the browser, open a new tab and go to `about://flags/#web-app-install-element`.
> 2. Set the Web App Install Element flag to **Enabled**.
> 3. Click the **Restart** button in the bottom right. The browser restarts.

Reviewer: ```suggestion
## Test the feature locally

To test the `<install>` element feature locally, in your browser only, use Microsoft Edge version 148 or later (or another Chromium-based browser) and enable the **Web App Install Element** experiment:

1. In the browser, open a new tab and go to `about://flags/#web-app-install-element`.
2. Enable the **Web App Install Element** flag.
3. Click the **Restart** button in the bottom right. The browser restarts.
```
